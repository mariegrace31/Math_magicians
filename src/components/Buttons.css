* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.Container {
  display: flex;
  justify-content: center;
  align-items: center;
}
