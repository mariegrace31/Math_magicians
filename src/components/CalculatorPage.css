* {
  padding: 0;
  margin: 0;
}

h2 {
  color: #fa7305;
  font-size: 29px;
  text-align: center;
  margin: 20px;
}
