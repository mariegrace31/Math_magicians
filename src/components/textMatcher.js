const textMatcher = (text) => text === '-' || text.includes('-');
export default textMatcher;
