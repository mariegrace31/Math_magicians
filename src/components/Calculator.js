import Buttons from './Button';
import OtherButtons from './OtherButton';
import Result from './Result';

function Calculator() {
  return (
    <div>
      <Result />
      <div>
        <div><Buttons /></div>
        <OtherButtons />
      </div>
    </div>
  );
}

export default Calculator;
