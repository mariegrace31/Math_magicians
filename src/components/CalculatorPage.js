import React from 'react';
import Navbar from './NavBar';
import Calculator from './Calculator';
import './CalculatorPage.css';

const CalculatorPage = () => (
  <div>
    <Navbar />
    <main>
      <h2>Let&apos;s do some Math!</h2>
      <Calculator />
    </main>
  </div>
);

export default CalculatorPage;
