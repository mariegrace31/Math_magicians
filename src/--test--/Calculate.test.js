import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import Buttons from '../components/Button';
import Calculator from '../components/Calculator';
import CalculatorPage from '../components/CalculatorPage';

import textMatcher from '../components/textMatcher';
import '@testing-library/jest-dom/extend-expect';

describe('Buttons component', () => {
 const mockHandleClick = jest.fn();
  it('should matche DOM Snapshot', async () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Buttons handleClick={mockHandleClick}/>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render buttons correctly', () => {
    const handleClick = jest.fn(); // Mocking the handleClick function
    render(<Buttons handleClick={handleClick} />);

    const buttonsNumber = ['7', '8', '9', '4', '5', '6', '1', '2', '3', '0'];
    buttonsNumber.forEach((buttonText) => {
      const button = screen.getByText(new RegExp(`\\b${buttonText}\\b`, 'i'));
      expect(button).toBeInTheDocument();
    });
    const operationButtons = ['\\+', '-', 'x', 'รท', '%', '\\+/-', '\\.'];
    operationButtons.forEach(() => {
      const button = screen.getByText(textMatcher, 'i');
      expect(button).toBeInTheDocument();
    });
    const acButton = screen.getByText('AC');
    expect(acButton).toBeInTheDocument();

    const equalButton = screen.getByText('=');
    expect(equalButton).toBeInTheDocument();
  });
  it('should call handleClick function on button click', () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Buttons handleClick={handleClick} />);
    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(handleClick).toHaveBeenCalledTimes(4);
    expect(handleClick).toHaveBeenCalledWith('1');
    expect(handleClick).toHaveBeenCalledWith('+');
    expect(handleClick).toHaveBeenCalledWith('2');
    expect(handleClick).toHaveBeenCalledWith('=');
  });
});
describe('Buttons component', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Calculator />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render without errors', () => {
    render(<Calculator />);
  });
});
describe('Buttons component', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <CalculatorPage />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render without errors', () => {
    render(
      <MemoryRouter>
        <CalculatorPage />
      </MemoryRouter>,
    );
  });
});
