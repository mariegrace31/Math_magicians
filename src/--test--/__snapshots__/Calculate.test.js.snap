// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons component should matche DOM Snapshot 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="right-button"
    onClick={[Function]}
    type="button"
  >
    ÷
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="right-button"
    onClick={[Function]}
    type="button"
  >
    ×
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="right-button"
    onClick={[Function]}
    type="button"
  >
    −
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="right-button"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="button zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="right-button"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`Buttons component should matche DOM Snapshot 2`] = `
<div
  className="Calc-container"
>
  <div
    className="Result-container"
  >
    <h1
      className="result"
    />
  </div>
  <div
    className="buttons"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="right-button"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="right-button"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="right-button"
      onClick={[Function]}
      type="button"
    >
      −
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="right-button"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="button zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="right-button"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Buttons component should matche DOM Snapshot 3`] = `
<div>
  <nav>
    <h1
      className="Title"
    >
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h2>
      Let's do some Math!
    </h2>
    <div
      className="Calc-container"
    >
      <div
        className="Result-container"
      >
        <h1
          className="result"
        />
      </div>
      <div
        className="buttons"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="right-button"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="right-button"
          onClick={[Function]}
          type="button"
        >
          ×
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="right-button"
          onClick={[Function]}
          type="button"
        >
          −
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="right-button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="button zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="right-button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </main>
</div>
`;
